tmux/2023-03-demo-6-vad-asr-tool copy.yml

```diff
@@ -0,0 +1,151 @@
+# Pass command line options to tmux. Useful for specifying a different tmux.conf.
+# tmux_options: -f ~/.tmux.mac.conf
+tmux_options: -f <%= ENV["ANGEL_WORKSPACE_DIR"] %>/tmux/tmux.conf
+
+windows:
+  - datahub: ros2 run ros_tcp_endpoint default_server_endpoint --ros-args
+      -r __ns:=${ROS_NAMESPACE}
+      -p ROS_IP:=0.0.0.0
+  - hl2ss_bridge: ros2 run angel_system_nodes hl2ss_ros_bridge --ros-args
+      -r __ns:=${ROS_NAMESPACE}
+      -p ip_addr:=${HL2_IP}
+      -p image_topic:=PVFramesBGR
+      -p hand_pose_topic:=HandJointPoseData
+      -p audio_topic:=HeadsetAudioData
+
+  # Visualize Images being output from the headset
+  - rqt_pv_images: rqt -s rqt_image_view/ImageView
+      --args ${ROS_NAMESPACE}/PVFramesBGR
+      --ros-args -p _image_transport:=raw
+
+  - object_detector: ros2 run angel_system_nodes object_detector_with_descriptors --ros-args
+      -r __ns:=${ROS_NAMESPACE}
+      -p image_topic:=PVFramesBGR
+      -p descriptor_topic:=ObjectDetections
+      -p min_time_topic:=ObjDetMinTime
+      -p torch_device:=cuda:0
+      -p detection_threshold:=0.05
+      -p object_vocab_list:=${MODEL_DIR}/fasterrcnn_label_list.txt
+      -p model_checkpoint:=${MODEL_DIR}/fasterrcnn_res101_vg.pth
+
+  - simple_2d_overlay: ros2 run angel_debug Simple2dDetectionOverlay --ros-args
+      -r __ns:=${ROS_NAMESPACE}
+      -p topic_input_images:=PVFramesBGR
+      -p topic_input_det_2d:=ObjectDetections
+      -p topic_output_images:=pv_image_detections_2d
+      -p filter_top_k:=5
+
+  - compress_image: ros2 run image_transport republish raw compressed --ros-args
+       -r __ns:=${ROS_NAMESPACE}
+       --remap in:=pv_image_detections_2d
+       --remap out/compressed:=pv_image_detections_2d/compressed
+
+  - activity_classifier: ros2 run angel_system_nodes uho_activity_detector --ros-args
+      -r __ns:=${ROS_NAMESPACE}
+      -p image_topic:=PVFramesBGR
+      -p hand_pose_topic:=HandJointPoseData
+      -p obj_det_topic:=ObjectDetections
+      -p torch_device:=cuda:1
+      -p det_topic:=ActivityDetections
+      -p min_time_topic:=ObjDetMinTime
+      -p frames_per_det:=32
+      -p object_dets_per_window:=2
+      -p buffer_max_size_seconds:=2.0
+      -p model_checkpoint:=${MODEL_DIR}/uho_checkpoint_20221022.ckpt
+      -p top_k:=10
+      -p overlapping_mode:=False
+  # Audio nodes
+  - vad: ros2 run angel_system_nodes voice_activity_detector --ros-args
+      -r __ns:=${ROS_NAMESPACE}
+      -p input_audio_topic:=HeadsetAudioData
+      -p output_voice_activity_topic:=DetectedVoiceData
+      -p vad_server_url:=http://communication.cs.columbia.edu:55667/vad
+      -p vad_cadence:=2
+      -p vad_margin:=0.25
+      -p max_accumulation_length:=10
+  - asr: ros2 run angel_system_nodes asr --ros-args
+      -r __ns:=${ROS_NAMESPACE}
+      -p audio_topic:=DetectedVoiceData
+      -p utterances_topic:=UtterancesTopic
+      -p asr_server_url:=http://communication.cs.columbia.edu:55667/asr
+      -p asr_req_segment_duration:=4.0
```

Comment:
@ovenmitt this  is the new addition  for running Voice Activity Detection prior to ASR.